core-bundle/src/Routing/ContentUrlGenerator.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing;
+
+use Contao\CoreBundle\Routing\Content\ContentUrlResolverInterface;
+use Contao\CoreBundle\Routing\Page\PageRegistry;
+use Contao\CoreBundle\Routing\Page\PageRoute;
+use Contao\Model;
+use Contao\PageModel;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\Mapping\MappingException;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Symfony\Component\Routing\Exception\ExceptionInterface;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RequestContextAwareInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class ContentUrlGenerator implements ResetInterface, RequestContextAwareInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private array $urlCache = [];
+
+    public function __construct(
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly PageRegistry $pageRegistry,
+        private readonly EntityManagerInterface $entityManager,
+        /**
+         * @var iterable<ContentUrlResolverInterface> $urlResolvers
+         */
+        private readonly iterable $urlResolvers,
+    ) {
+    }
+
+    /**
+     * @throws ExceptionInterface
+     */
+    public function generate(object $content, array $parameters = [], array $optionalParameters = []): string
```

Comment:
I still think this should be 

`public function generate(object $content, ContentUrlParameters $parameters): string`

With helpers just like you did on the `ContentUrlResult` which is beautiful 😊  Something tells me that we'll end up needing more info than just those 2 arguments in the future 🙈

Reply:
This is the only thing left that prevents me from approving this PR. I really think we should address this because I think it's bad design otherwise.
If this was converted to a value object and tests added, I would happily approve this PR.

Reply:
I would prefer this:

```php
public function generate(object $content, ContentUrlParameters|null $parameters = null, int $referenceType = self::ABSOLUTE_URL): string
```

It would follow the Symfony `UrlGeneratorInterface` and – at least in Contao 5 – the `PageModel::getFrontendUrl()` method still uses `ABSOLUTE_PATH` instead of `ABSOLUTE_URL`.